docs/user_guide/discretisation/EqualWidthDiscretiser.rst

```diff
@@ -5,77 +5,98 @@
 EqualWidthDiscretiser
 =====================
 
-The :class:`EqualWidthDiscretiser()` sorts the variable values into contiguous intervals
-of equal size. The size of the interval is calculated as:
+The :class:EqualWidthDiscretiser() divides continuous numerical variables into intervals of equal width, calculated using the formula:
 
 ( max(X) - min(X) ) / bins
 
-where bins, which is the number of intervals, should be determined by the user. The
-interval limits are determined using `pandas.cut()`.
+where bins is the number of intervals specified by the user. This discretisation is achieved through the pandas.cut() function.
 
-**A note on number of intervals**
+Discretization is a common data preprocessing technique used in data science. It's also known as data binning (or simply `binning`).
 
-Common values are 5 and 10. Note that if the variable is highly skewed or not continuous
-smaller intervals maybe required. Otherwise, the transformer will introduce np.nan.
+Advantages and Limitations
+--------------------------
 
-The :class:`EqualWidthDiscretiser()` works only with numerical variables. A list of
-variables to discretise can be indicated, or the discretiser will automatically select
-all numerical variables in the train set.
+Advantages
+~~~~~~~~~~
 
-**Example**
+Some advantagers of equal width binning:
 
-Let's look at an example using the House Prices Dataset (more details about the
-dataset :ref:`here <datasets>`).
+- **Algorithm Efficiency:** Enhances the performance of data mining and machine learning algorithms by providing a simplified representation of the dataset.
+- **Outlier Management:** Efficiently mitigates the effect of outliers by grouping them into the extreme bins, thus preserving the integrity of the main data distribution.
+- **Data Smoothing:** Helps smooth the data, reduces noise, and improves the model's ability to generalize.
 
-Let's load the house prices dataset and  separate it into train and test sets:
+Limitations
+~~~~~~~~~~~
+
+On the other hand, :class:`EqualWidthDiscretiser()` can lead to a loss of information by aggregating data into broader categories. This is particularly concerning if the data in the same bin has predictive information about the target.
+
+Let's consider a binary classifier task using a decision tree model. A bin with a high proportion of both categories would potentially impact the model's performance in this scenario.
+
+Notes
+-----
+
+`EqualWidthDiscretiser` expects a `pandas.DataFrame` and works only with numerical variables. The user can specify the variables to be discretized. Otherwise, `EqualWidthDiscretiser` will automatically infer the data types to compute the interval limits for all numeric variables.
+
+**Optimal number of intervals:** With `EqualWidthDiscretiser`, the user defines the number of bins. Smaller intervals may be required if the variable is highly skewed or not continuous. Otherwise, the transformer will introduce `numpy.nan`.
```

Comment:
did you test this and you saw that the transformer introduces numpy nan? I don't remember this behaviour.

Reply:
Actually, I don't! I kept this information from the original user guide page under the topic "A note on number of intervals". It is currently stated for both `EqualFrequencyDiscretiser` and `EqualWidthDiscretiser`. I think we can remove this statement since I was not able to reproduce this behavior. What do you think?

Reply:
Sounds good. Remove it then! Sometimes I hallucinate, like ChatGPT.